frontend/web/src/features/guestdashboard/GuestReviews.js

```diff
@@ -1,213 +1,199 @@
-                        'Content-type': 'application/json; charset=UTF-8',
-                    }
-                });
-                if (!response.ok) {
-                    throw new Error(`HTTP error! Status: ${response.status}`);
-                }
-                const updatedReviews = reviews.filter(r => r["reviewId "] !== reviewId);
-                setReviews(updatedReviews);
-            } catch (error) {
-                console.error(error);
-            }
+  const [reviews, setReviews] = useState([]);
+  const [receivedReviews, setReceivedReviews] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading2, setIsLoading2] = useState(true);
+  const [userId, setUserId] = useState(null);
+  const navigate = useNavigate();
+
+  // Get user once & redirect if not logged in
+  useEffect(() => {
+    (async () => {
+      try {
+        const userInfo = await Auth.currentUserInfo();
+        const sub = userInfo?.attributes?.sub;
+        if (sub) setUserId(sub);
+        else navigate("/login");
+      } catch (err) {
+        console.error("Auth/currentUserInfo error:", err);
+        navigate("/login");
+      }
+    })();
+  }, [navigate]);
+
+  // Fetch my reviews (written by me)
+  useEffect(() => {
+    if (!userId) return;
+
+    const retrieveReviews = async () => {
+      setIsLoading(true);
+      try {
+        const res = await fetch(
+          "https://arj6ixha2m.execute-api.eu-north-1.amazonaws.com/default/FetchReviews",
+          {
+            method: "POST",
+            body: JSON.stringify({ userIdFrom: userId }),
+            headers: { "Content-type": "application/json; charset=UTF-8" },
+          }
+        );
+        if (!res.ok) throw new Error(`HTTP ${res.status}`);
+        const data = await res.json();
+        setReviews(Array.isArray(data) ? data : []);
+      } catch (e) {
+        console.error("FetchReviews error:", e);
+        setReviews([]);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    retrieveReviews();
+  }, [userId]);
+
+  // Fetch reviews I received
+  useEffect(() => {
+    if (!userId) return;
+
+    const retrieveReceivedReviews = async () => {
+      setIsLoading2(true);
+      try {
+        const res = await fetch(
+          "https://arj6ixha2m.execute-api.eu-north-1.amazonaws.com/default/FetchReceivedReviews",
+          {
+            method: "POST",
+            body: JSON.stringify({ itemIdTo: userId }),
+            headers: { "Content-type": "application/json; charset=UTF-8" },
+          }
+        );
+        if (!res.ok) throw new Error(`HTTP ${res.status}`);
+        const data = await res.json();
+        setReceivedReviews(Array.isArray(data) ? data : []);
+      } catch (e) {
+        console.error("FetchReceivedReviews error:", e);
+        setReceivedReviews([]);
+      } finally {
+        setIsLoading2(false);
+      }
+    };
+
+    retrieveReceivedReviews();
+  }, [userId]);
+
+  // Delete a review I wrote
```

Comment:
Again, saying Delete a review I wrote is pretty unclear for a other developer to who "I" is actually referred to.

```suggestion
  // Deletes a user's written review
```